llrt_core/src/modules/http/headers.rs

```diff
@@ -122,16 +174,72 @@ impl Headers {
     }
 
     pub fn from_map(map: BTreeMap<String, Coerced<String>>) -> Self {
-        let headers: BTreeMap<String, String> = map
-            .into_iter()
-            .map(|(key, value)| (key.to_lowercase(), value.to_string()))
-            .collect();
-        Self { headers }
+        let mut headers = Headers::default();
+
+        for (key, value) in map {
+            headers.insert_header_value(key, value.to_string(), true);
+        }
+        headers
+    }
+
+    fn insert_header_value(&mut self, key: String, value: String, appending: bool) {
+        let key = key.to_lowercase();
+
+        if let Some(header_value) = self.headers.get_mut(&key) {
+            match header_value {
+                HeaderValue::Single(existing_value) => match appending {
+                    true => {
+                        *header_value =
+                            HeaderValue::Single(format!("{}, {}", existing_value, value))
```

Comment:
We do not believe it is necessary to assume string concatenation with ";" since the set-cookie is always determined to be a MultipleValue when it is first registered in the header.

Reply:
Yes for `set-cookie` but not for `cookie` header

Reply:
Yes for `set-cookie` but not for `cookie` header

Reply:
Oh, that's right. I was omitting a consideration.